### ISSUE TITLE
Cant install Pdf.JS on React Native

### ISSUE THREAD

**Issue by diogoviannaaraujo** [mozilla/pdf.js#8774]
Configuration:
- MacOS and Node 8.x
- React Native

Steps to reproduce the problem:
- Start a react native project
- Yarn add pdfjs-dist
- Try to import pdfjs-dist

You will get a error like this 

> Unable to resolve module `./zlib_bindings` ./node_modules/zlib/lib/zlib_bindings doesn't exist

**Comment by timvandermeij** [comment]
`zlib` should be available in Node.js. IIRC `zlib` is only used for the experimental SVG back-end for optimizations, so I'm not sure why you're getting this. Does `npm install zlib` work? If it does, perhaps we need to add it to the `devDependencies` in `package.json`?

**Comment by yurydelendik** [comment]
Please provide complete example.

**Comment by yurydelendik** [comment]
Actually, I'll close it won't fix. Even if we will solve zlib_binding, there is no guarantee rest of PDF.js will work -- PDF.js need "native" browser's DOM to work.

**Comment by diogoviannaaraujo** [comment]
React Native code does not run on a full node.js environment, so zlib is not available by default. 

If I just add pdfjs-dist to package.json it goes to a zlib com not be found error.
Then I added https://www.npmjs.com/package/zlib

And got that error,
I think it should threat React Native as a browser js not as a node

**Comment by flieks** [comment]
is it possible to use webGL to achieve this ?

**Comment by yurydelendik** [comment]
> is it possible to use webGL to achieve this ?

Anything is possible -- you just need to implement logic similar to what src/display/canvas.js or src/display/svg.js implements.

**Comment by flieks** [comment]
Haha thanks you but those files are 2000 lines of code and my webGL knowledge is nihil :)

Also pdfjs doesn't work well with expo yet (zlib). Neither webview 

https://github.com/expo/expo-sdk/issues/82

**Comment by diogoviannaaraujo** [comment]
Latter next month I may have to dig in a way to use pdf.js with expo (and react-native).
I think using a abstracted dom or a web view is not the way, maybe gl, svg or canvas, still have to take some time to understand better what pdf.js is capable.
About zlib and other node libraries, I’ve found out you can go around that by implementing some aliases on babel and installing some npm packages. But there is a lot of going around to do and a lot of bugs to catch.

**Comment by flieks** [comment]
yes, Expo doesn't have way to show PDF right now and neither has PDF.js. 
I will use https://github.com/wonday/react-native-pdf and will have to detach from Expo i fear. Don't really want to detach but there is no other way

**Comment by agilgur5** [comment]
Was looking into this myself, so figured I might leave some notes for future readers.
Initially I found https://github.com/iddan/react-native-canvas which could move a step forward in at least getting rendering to work. It uses a WebView under the hood.
When looking into WebView support, I did find this issue: https://github.com/mozilla/pdf.js/issues/7157#issuecomment-394066217 , and the comment which points to https://github.com/xcarpentier/rn-pdf-reader-js , which was actually built for Expo. It similarly uses a WebView under the hood (and uses `react-pdf` instead of `pdf.js` directly)
